File: algorithms/Javascript/insertionSort.js
// O(n^2) Time | O(1) Space
function insertionSort(array) {
	for (let i = 0; i < array.length; i++) { // loop through array
		let j = i;
		while (j > 0 && array[j] < array[j - 1]) { // j loops backwards to start from i
			const temp = array[j - 1]; // if num is greater than num before it, swap them
			array[j - 1] = array[j];
			array[j] = temp;
			j--;
		}
	}
	return array;
}

File: algorithms/Python/insertionSort.py
# O(n^2) Time | O(1) Space
def insertionSort(array):
    for i in range(1, len(array)):
        j = i
        while j > 0 and array[j] < array[j - 1]: # j iterates backwards from i to start of array
            array[j], array[j - 1] = array[j - 1], array[j]
            j -= 1  # if num is greater than num right before it, swap them
    return array
